File: backend/src/main/java/es/udc/tfgproject/backend/model/services/CompanyService.java
package es.udc.tfgproject.backend.model.services;

import es.udc.tfgproject.backend.model.entities.Company;
import es.udc.tfgproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.tfgproject.backend.model.exceptions.WrongUserException;

public interface CompanyService {

	Company addCompany(Long userId, String name, int capacity, Boolean reserve, Boolean homeSale, int reservePercentage,
			Long companyCategoryId) throws InstanceNotFoundException;

	Company modifyCompany(Long userId, Long companyId, String name, int capacity, Boolean reserve, Boolean homeSale,
			int reservePercentage, Long companyCategoryId) throws InstanceNotFoundException, WrongUserException;

	void deregister(Long userId, Long companyId) throws InstanceNotFoundException, WrongUserException;

}

File: frontend/src/backend/companyService.js
import {config, appFetch} from './appFetch';

export const addCompany = (name, capacity, reserve, homeSale, reservePercentage, companyCategoryId, 
    onSuccess, onErrors) =>
    appFetch(`/companies/add`,
        config('POST', {name, capacity, reserve, homeSale, reservePercentage, companyCategoryId}),
        onSuccess, onErrors);

export const modifyCompany = (company, onSuccess, onErrors) =>
    appFetch(`/companies/${company.id}`, config('PUT', company),
        onSuccess, onErrors);

export const deregister = (id, onSuccess, onErrors) =>
    appFetch(`/companies/${id}/deregister`, 
        config('POST'), onSuccess, onErrors);

export const findAllCompanyCategories = (onSuccess) => 
appFetch('/companies/categories', config('GET'), onSuccess);
    